lib/dry/auto_inject/strategies/kwargs.rb

```diff
@@ -72,7 +72,7 @@ def define_initialize_with_splat(super_parameters)
               else
                 super_kwargs = slice_kwargs.(kwargs, super_parameters)
 
-                if super_kwargs.any?
+                if super_kwargs.any? && super_parameters.keyword_names.any?
```

Comment:
Does it allocate a new array every time when asking for `keyword_names`? If yes, then it would be good to memoize it

Reply:
As I read it, `keyword_names` [are already memoized](https://github.com/dry-rb/dry-auto_inject/blob/master/lib/dry/auto_inject/method_parameters.rb#L48).